.gitignore

```diff
@@ -34,6 +34,7 @@
 
 # the build directory
 build/
+/build-python/
```

Comment:
I agree that placing the build library in the project root might not be the best approach.

But the change was made to be able to build both the main C++ library plus tests for it and python binding simultaneously. Or at least so we do not have to wait for the main library to be built and tested in order to start building the python binding which takes a considerable time when developing. What I want is the following commands to:
- build C++ library
- build C++ library plus tests, examples, integrations tests
- run integration tests
- build Python binding
- build Python binding and run Python binding tests

However, I do not want to run the command to build library + tests and examples, wait for it to finish, and then run building of python binding for tests and wait again. Then change something and run C++ build and have to wait again for running the Python bindings again only after the C++ build finishes.

Ultimately, what I want is to have a command line flag or some easy to use and set parameter for each type of build: release/debug/release with debug information/python binding which would specify where (in which folder) to build the library. So that I can easily run in two terminals immediately one after the other: `run_tests_build_examples`, `run_python_tests`. If I do it now, the two GCC instances rewrite each other's files and the whole build fails.

We could put the `build-python` folder inside `build/` as `build/bindings/python`, for example. Or have a structure like this:
```
build/
    - bindings
        - python # Python bindings build
    - library
        - debug # C++ lib debug build
        - release # C++ lib release build
        - release_debug_info # C++ lib release with debug information build
```
When developing, I need to see the results of each of these four builds (both compilation and tests), without having to rebuild the entire library again for each one because the previous build compiled the library with different options.

Reply:
Hm, it took me some time, but I think I finally get it. You mean, that your main problem is, that you are e.g. (illustrative; don't take me literally) working on some problem, and you want to have library compiled for DEBUG, and binding built for RELEASE, but if these shared the `build` (as it did so far), they would clash. I think you got me with the simultaneously (I interpreted it as "parallelly", which I though was quite weird).

Such interpretation is fine by me. Since I mostly worked on binding only, I was fine with "reusing" the build both for binding and libmata.

Reply:
You are right, but that is only a part of the problem. Next, I also want to do the same thing as for DEBUG for release (and measure performance testing) and for release with debug info (and run tests).

Moreover, what I want is to be able to do some change and decide which of these commands I need to run after this change. Then run the following commands, each in their own terminal window, *simultaneously*. That is, start all of them in like two seconds, go do some other stuff and then return and have results for all of these builds. Therefore, not only would release and debug overwrite each other, they also overwrite each other because they are (ideally, how I would want them to) run in parallel, indeed.

After discussing this with @jurajsic, we came to the conclusion that I can already set up the library builds exactly as described above by specifying a command line flag `BUILD_DIR` for `cmake`. However, I cannot do that for Python binding yet. Therefore, the solution for now would be to move the default Python binding build directory to `build/python` or `build/bindings/python`. But, for this PR, we will revert these changes, build Python to the default `build/` folder and I will try to make a PR in the future allowing to specify a custom build folder for Python binding by a command line flag, the same way as for C++ library builds.

Reply:
Mata for python bindings should be built in `build/` again.